src/fuse_pointwise.cpp

```diff
@@ -183,10 +182,11 @@ static bool find_pointwise_modules(module& m)
             continue;
         auto input = *it;
 
-        auto new_inputs = append_pointwise_module(input, ins);
-        m.replace_instruction(input, input->get_operator(), new_inputs, input->module_inputs());
-        m.replace_instruction(ins, input);
-        m.move_instruction(input, ins);
+        auto fused = append_pointwise_module(input, ins);
+        auto name  = fused.mod.name();
+        mpm.rename_module(name, name + ":" + ins->module_inputs().front()->name() + "-deleted");
```

Comment:
why `name+:` is prepended on deleted module ?

Reply:
why does it require renaming ? It's appending current pointwise module to previous pointwise module. Fused module would take name of previous pointwise module. So why is there a need to rename module?

Reply:
Because the fused module uses the same name as the first pointwise, so it needs to rename it so we can add a module with this name, we can remove the module yet until DCE removes the unused function, so thats why we rename it to something that shouldn't conflict with other module names.

Reply:
i don't understand why it is necessary to rename it. `ins` is being replaced on line 189 and that would remove all its' usages and then DCE runs just after `append_pointwise`

Reply:
If I replace `input`, then it will use the inputs to `ins` which some could come after the `input` instruction, which will make an invalid module.

Reply:
> which some could come after the input instruction, which will make an invalid module. 

Looks like current logic on develop branch would fail in that case. I wonder how come current logic on develop didn't stumble into that case.

Reply:
On develop, it replaced the input instruction, then replaced the usage of `ins`, and then moved it before the `ins`. However, we would still need to rename the module if we did this.

The reason why we dont need to rename it on develop was because we werent checking that the input types matched. So if we replace `input` it can create a different output type(like when we fuse a `convert`). This will still be used by `ins`, so a recompute shape is called, but there is no mismatch error(even though the types no longer match). 

Instead of going through this dance of replace input, then `ins` and move it. I just replace `ins` instead.